### ISSUE TITLE
Windows 11 Context Menu

### ISSUE THREAD

**Issue by minzdrav** [gitextensions/gitextensions#9824]
### Feature description

Hi GitExtensions team
Could you please add support for new context menu in Windows 11? At least one menu item: "Open in Git Extensions".
Additional info about new context menu: https://blogs.windows.com/windowsdeveloper/2021/07/19/extending-the-context-menu-and-share-dialog-in-windows-11/
Thank you!
![image](https://user-images.githubusercontent.com/1764392/150083447-8812962f-f523-49bf-9ae0-fcd980597a8e.png)

### Environment

- GitExtensions version: future
- GIT version: any
- OS version: Windows 11
- .NET version: any

**Comment by RussKie** [comment]
Setting is a up for grabs. A PR with a fix is welcome.

**Comment by mika76** [comment]
This would be awesome!

**Comment by jmg2k** [comment]
@RussKie @mstv is there any documentation on how the signing process is done for GitExtensions? I finally snapped about this annoyance (as GitExtensions is THE ONLY app not yet in the new context menu, which I use daily) and so, spent the last night on it and got to a generally working state with a self-signed certificate now. The main thing left to do is incorporation into the installer (well, aside from testing and clean-up...).

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/9e2205ad-a30d-40de-969d-ce7f517a14df" />

**Comment by mstv** [comment]
Only CI builds of release branches can be signed. An administrator needs to approve the signing request at SignPath.
Then, all binaries in the installer and in the portable package are signed by SignPath.

**Comment by jmg2k** [comment]
So I won't have to do anything when adding a new DLL (I tried extending the existing one first, but deemed it easier and less invasive / fault tolerant to not touch the existing one) and the necessary MSIX file, and as long as both finally exist in artifacts/Release/publish/GitExtensions, they'll be signed?

If you would endulge me, I also encountered an issue, in that the MSIX can only be deployed per user. So an install action for the MSI setup would only activate the extension for the user installing it. For me and my collegues (and probably many other developers) this would be totally fine as there's just one user per machine. But it's not quite "clean". I wonder, would you accept this shortcoming in a PR? Or should it not be part of the MSI setup and instead get enable/disable buttons somewhere in the GUI options dialog, like WinMerge does it, and each user has to click there manually?

Firing off the MSIX deployment command in the MSI setup is easy enough, for the current user activation. I could do modifications in C# too, but am not familiar at all with the code base and would like to get a PR for the initial feature with a minimum viable solution out there before the job steals all the time. The GUI options like in WinMerge could later be added (maybe by someone familiar with the code base) to make it usable for multiple user machines (aside from manually installing the MSIX directly via PowerShell, which at least power users will be able to do).

So... Some guidance on how to proceed with this please 🙂

**Comment by mstv** [comment]
> So I won't have to do anything when adding a new DLL (I tried extending the existing one first, but deemed it easier and less invasive / fault tolerant to not touch the existing one) and the necessary MSIX file, and as long as both finally exist in artifacts/Release/publish/GitExtensions, they'll be signed?

Yes, every DLL in this folder is packaged and the package is signed - but only when an official release candidate is built.
In order to add the DLL to the `.msi` file, a few lines need to be added to the installer config.

A draft pull request eases the discussion and enables first (manual) tests.

If the per-user registration does not require admin privileges, the settings page option might suffice. For the portable package, MSIX is not an option anyway.
Though enterprise admins prefer automated installs per machine. But this is no blocker for a PR.
The buttons on the settings page could be extended in a follow-up, too.

<img width="710" height="607" alt="Image" src="https://github.com/user-attachments/assets/237d1ee7-c7fa-4cc6-aa70-73399f4288fb" />

---

In order to avoid merge conflicts, file your sign-off to `contributors.txt` as separate PR, please.

**Comment by gerhardol** [comment]
> So I won't have to do anything when adding a new DLL (I tried extending the existing one first, but deemed it easier and less invasive / fault tolerant to not touch the existing one) and the necessary MSIX file, and as long as both finally exist in artifacts/Release/publish/GitExtensions, they'll be signed?

dlls must be added in setup/installer/Product.wxs for the installer to build correctly.
Signing is "automatic" of all files.

**Comment by jmg2k** [comment]
> The buttons on the settings page could be extended in a follow-up, too.

Oh wow, I just notice I've ever only got into the Git user settings page for name, email and diff tool, pretty much using GitExtensions with it's default. Never noticed there's already configuration options 😅. In that case I'll probably be able to add them easily enough too (though only enable/disable, the actual commands are currently static).

> A draft pull request eases the discussion and enables first (manual) tests.

I'll add a script to manually install with an on the fly generated certificate for easy testing/development and then make the draft PR soon, with the current state it's in 🙂

**Comment by jmg2k** [comment]
> In order to avoid merge conflicts, file your sign-off to contributors.txt as separate PR, please.

Made that in #12776.

Draft PR #12777 containing the current state is open. CodeFactor complained about an insecurity in a convenience script for local development, but further discussions will probably then be happening over there...
